biz.aQute.repository/src/aQute/bnd/jpm/Repository.java

```diff
@@ -1606,6 +1606,7 @@ protected void fireBundleAdded(File file) throws IOException {
 	public void setRegistry(Registry registry) {
 		this.registry = registry;
 		this.httpClient = registry.getPlugin(HttpClient.class);
+		this.httpClient.setRegistry(registry);
```

Comment:
Hmm, right, that could be trouble, I checked and turns out this isn't even needed, sending a new commit.